seatunnel-plugin-discovery/src/main/java/org/apache/seatunnel/plugin/discovery/AbstractPluginDiscovery.java

```diff
@@ -439,10 +438,10 @@ public boolean accept(File pathname) {
             if (targetPluginFiles.length == 1) {
                 pluginJarPath = targetPluginFiles[0].toURI().toURL();
             } else {
+                PluginType type = PluginType.valueOf(pluginType.toUpperCase());
+                String targetPlugin = getTargetPluginName(pluginIdentifier, type);
                 pluginJarPath =
-                        findMostSimlarPluginJarFile(targetPluginFiles, pluginJarPrefix)
-                                .toURI()
-                                .toURL();
+                        selectPluginJar(targetPluginFiles, targetPlugin, pluginName, type).get();
```

Comment:
The `pluginJarPrefix` equals `targetPlugin`. So we can just use `pluginJarPrefix`.